### DISCUSSION TITLE
Index types of object's property which is array

### DISCUSSION THREAD

**Discussion by hrishikesh-k** [sinclairzx81/typebox#793] [Q&A] [answered]
I have a type like:

```ts
const t1 = Object({
  f: Array(Object({
    id: Integer()
  }))
})
```

I wish to add another type that has a type of singular item of that array, so something like:

```ts
const t2 = Object({
  p: Index(t1, ['f']['number'])
})
```

But it does not working as it gives me an error:

```
Cannot read properties of undefined (reading 'map')
  at IndexFromPropertyKeys (/node_modules/@sinclair/typebox/build/import/type/indexed/indexed.mjs:59:14)
  at FromSchema (/node_modules/@sinclair/typebox/build/import/type/indexed/indexed.mjs:63:28)
  at Index (/node_modules/@sinclair/typebox/build/import/type/indexed/indexed.mjs:71:27)
```

I have multiple ways to solve it like declaring a separate const and using that in both types or declaring t2 first and then using that as the array item in t1. But I'm simply curious if it's possible to solve this issue in my current setup.

**Comment by sinclairzx81** [comment, marked as answer]
@hrishikesh-k Hi,

You can implement this type in the following way

[TypeScript Link Here](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgFQJ5gKYBo4GUYCGMwAxnAL5wBmUEIcA5AAIDOwAdiQDYHBQD0MdBgBGEAB4MAUFP784QzCjgBeRLPlxqALjgAKJMAAmu9gFcQIjFAoBKANoBdDRRkkI7FvGSqUwgHQA8iIAVhgkMAZSWlS6aJj+AIJQUASoevEYQaHhkQjRWnDGcQEAkuwwGADm1nq2BeS29Y0ycgrCcAAavvltWmBx9gxUDI725pbWzm3kbh5eXb6Z2WERUYVwA34J5UYY4hll7HsHyDjLADLAlalcesMMTecBAHIWVlB1zfVAA)

```typescript
import { Type, Static } from '@sinclair/typebox'

// type T = {
//   f: ({ id: number })[]
// }

const T = Type.Object({
  f: Type.Array(Type.Object({
    id: Type.Integer()
  }))
})

// type X = {
//   p: T['f'][number]
// }

const X = Type.Object({
    p: Type.Index(Type.Index(T, Type.Literal('f')), Type.Number())
})

```
Unfortunately, TypeBox can't support the `['f'][number]` syntax, so to access deeper nested properties, you will need to use Index multiple times. Also note, for Arrays, you can pass `Type.Number()` to access the element type.

Hope this helps!
S

**Reply by hrishikesh-k** [reply]
That was it, thank you so much!
